buildserver/build_server.go

```diff
@@ -327,10 +349,27 @@ func (h *BuildHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jso
 			}
 		}
 		rewriteURIFromClient := func(uri lsp.DocumentURI) lsp.DocumentURI {
-			if !strings.HasPrefix(string(uri), "file:///") {
-				return uri // refers to a resource outside of this workspace
+			var path string
+			if h.clientUsesFileSchemeWithinWorkspace {
+				if !strings.HasPrefix(string(uri), "file:///") {
+					return uri // refers to a resource outside of this workspace
+				}
+				path = strings.TrimPrefix(string(uri), "file://")
+			} else {
+				currentURL, err := url.Parse(string(uri))
+				if err != nil {
+					return uri
+				}
+				originalRootURI, err := url.Parse(string(h.init.OriginalRootURI))
```

Comment:
minor: Anyway we can avoid parsing this here? This function can be called a lot, especially for large results.

Reply:
✅ [Do not parse OriginalRootURI so often](https://github.com/sourcegraph/go-langserver/pull/373/commits/f1f8c3d5674c2aa4b2fe5ca267fbd5b1d74be42a)